mlflow/store/dbmodels/models.py

```diff
@@ -0,0 +1,190 @@
+
+        config[k] = obj
+    return base(**config)
+
+
+class SqlExperiment(Base):
+    __tablename__ = 'experiments'
+    __entity__ = Experiment
+    __properties__ = Experiment._properties()
+    experiment_id = Column(Integer, primary_key=True)
+    is_deleted = Column(Boolean, default=False)
+    name = Column(String(256), unique=True, nullable=False)
+    artifact_location = Column(Text, nullable=True)
+    lifecycle_stage = Column(Integer, default=Experiment.ACTIVE_LIFECYCLE)
+
+    __table_args__ = (
+        CheckConstraint(
+            lifecycle_stage.in_(ExperimentLifecycleStages), name='lifecycle_stage'),
+    )
+
+    def __repr__(self):
+        return '<SqlExperiment ({}, {})>'.format(self.experiment_id, self.name)
+
+    def to_mlflow_entity(self):
+        return _create_entity(Experiment, self)
+
+
+class SqlRunTag(Base):
+    __tablename__ = 'run_tag'
+    __entity__ = RunTag
+    __properties__ = RunTag._properties()
+    id = Column(Integer, primary_key=True)
+    key = Column(Text, nullable=False)
+    value = Column(Text, nullable=True)
+    run_uuid = Column(Integer, ForeignKey('run.run_uuid'))
+    run = relationship('SqlRun', backref=backref('tags', cascade='all,delete'))
+
+    def __repr__(self):
+        return '<SqlRunTag({}, {})>'.format(self.key, self.value)
+
+    def to_mlflow_entity(self):
+        return _create_entity(RunTag, self)
+
+
+class SqlMetric(Base):
+    __tablename__ = 'metric'
+    __entity__ = Metric
+    __properties__ = Metric._properties()
+    key = Column(Text, primary_key=True)
+    value = Column(Float, nullable=False)
+    timestamp = Column(Integer, default=int(time.time()), primary_key=True)
+    run_uuid = Column(Integer, ForeignKey('run.run_uuid'), primary_key=True)
+    run = relationship('SqlRun', backref=backref('metrics', cascade='all,delete'))
+
+    def __repr__(self):
+        return '<SqlMetric({}, {})>'.format(self.key, self.value)
+
+    def to_mlflow_entity(self):
+        return _create_entity(Metric, self)
+
+
+class SqlParam(Base):
+    __tablename__ = 'param'
+    __entity__ = Param
+    __properties__ = Param._properties()
+    key = Column(Text, primary_key=True)
+    value = Column(Text, nullable=False)
+    run_uuid = Column(Integer, ForeignKey('run.run_uuid'), primary_key=True)
+    run = relationship('SqlRun', backref=backref('params', cascade='all,delete'))
+
+    def __repr__(self):
+        return '<SqlParam({}, {})>'.format(self.key, self.value)
+
+    def to_mlflow_entity(self):
+        return _create_entity(Param, self)
+
+
+class SqlRun(Base):
+    __tablename__ = 'run'
+    __entity__ = Run
+    __properties__ = Run._properties()
+
+    id = Column(Integer, primary_key=True)
+    is_deleted = Column(Boolean, default=False)
+    run_uuid = Column(String(16), default=generate_uuid, unique=True, nullable=False)
+    name = Column(Text, unique=True)
+    source_type = Column(Integer, default=SourceType.LOCAL)
```

Comment:
I am concerns about some of the datatype and sizes. Let's avoid using `Text` type.
Also as noted above for enums might be worth converting to strings.
timestamps as `BigInteger`

```
    name = Column(String(200), unique=True)
    source_type = Column(String(20), default=enumToString(SourceType.LOCAL))
    source_name = Column(String(500))
    entry_point_name = Column(String(50))
    user_id = Column(String(50), default=_get_user_id(), nullable=False)
    status = Column(String(20), default=enumToString(RunStatus.SCHEDULED))
    start_time = Column(BigInteger, default=int(time.time()))
    end_time = Column(BigInteger, nullable=True, default=None)
    source_version = Column(String(20))
    lifecycle_stage = Column(String(20), default=enumToString(RunInfo.ACTIVE_LIFECYCLE))
    artifact_uri = Column(String(20), default=None)
```